ratatui/benches/main/constraints.rs

```diff
@@ -0,0 +1,59 @@
+use std::hint::black_box;
+use std::rc::Rc;
+
+use criterion::{Criterion, criterion_group};
+use ratatui::layout::Constraint::{Fill, Length, Max, Min, Percentage, Ratio};
+use ratatui::layout::{Direction, Layout, Rect};
+
+const SPLIT_BY: u16 = 10;
+
+fn generate_layout(constraint: &str, area: Rect) -> Rc<[Rect]> {
+    Layout::default()
+        .direction(Direction::Vertical)
+        .constraints({
+            let mut cs = Vec::new();
+            for _ in 0..SPLIT_BY {
+                match constraint {
+                    "Fill" => {
+                        cs.push(Fill(1));
+                    }
+                    "Length" => {
+                        cs.push(Length(area.width / SPLIT_BY));
+                    }
+                    "Max" => {
+                        cs.push(Max(area.width / SPLIT_BY));
+                    }
+                    "Min" => {
+                        cs.push(Min(area.width / SPLIT_BY));
+                    }
+                    "Percentage" => {
+                        cs.push(Percentage(100 / SPLIT_BY));
+                    }
+                    "Ratio" => {
+                        cs.push(Ratio(1, SPLIT_BY.into()));
+                    }
+                    _ => todo!(),
+                }
+            }
+            cs
+        })
```

Comment:
Generally with benchmarking the idea is to ensure that you only have code which is actually being benchmarked in the code that runs for each iteration. This part of the code is probably fairly fast, but it shouldn't be part of each layout call as it's more related to choosing what to test than it is actually running the test.